packages/network-controller/src/NetworkController.ts

```diff
@@ -481,6 +539,131 @@ export class NetworkController extends BaseControllerV2<
       });
     }
   }
+
+  /**
+   * Adds a network configuration if the rpcUrl is not already present on an
+   * existing network configuration. Otherwise updates the entry with the matching rpcUrl.
+   *
+   * @param networkConfiguration - The network configuration to add or, if rpcUrl matches an existing entry, to modify.
+   * @param networkConfiguration.rpcUrl -  RPC provider url.
+   * @param networkConfiguration.chainId - Network ID as per EIP-155.
+   * @param networkConfiguration.ticker - Currency ticker.
+   * @param networkConfiguration.nickname - Personalized network name.
+   * @param networkConfiguration.rpcPrefs - Personalized preferences (i.e. preferred blockExplorer)
+   * @param options - additional configuration options.
+   * @param options.setActive - An option to set the newly added networkConfiguration as the active provider.
+   * @param options.referrer - The site from which the call originated, or 'metamask' for internal calls - used for event metrics.
+   * @param options.source - Where the upsertNetwork event originated (i.e. from a dapp or from the network form)- used for event metrics.
```

Comment:
Nit: Space missing after closing paren, maybe a typo
```suggestion
   * @param options.source - Where the upsertNetwork event originated (i.e. from a dapp or from the network form) - used for event metrics.
```

Reply:
fixed here: https://github.com/MetaMask/core/pull/1064/commits/0124a625a57d81ae00c1dcaa03e880c62f6ff14e#diff-ea0aa37793484e278bd59970b7a6ce8380b75b723b34612081d402fb66261cfdR556